### DISCUSSION TITLE
v0.22.0 on Windows 10 took minutes to reload

### DISCUSSION THREAD

**Discussion by adhihargo** [Kludex/uvicorn#1977] [Potential Issue] [answered]
Uvicorn used to work fine in my dev laptop running Windows 10, but now every time there's a modified change, it stuck in this line:

`WARNING:  WatchFiles detected changes in 'main.py'. Reloading...`

... for nearly 5 minutes before actually reloading the files (printing `Shutting down` in the next line, etc.). During that time, the API server part works normally but serving the older version before file modification..

Downgraded to v0.21.1, and reloading works instantly again.

**Comment by 3to1null** [comment]
I have the exact same issue. Upgrading to v0.22.0 makes reloading take ages, even with `timeout_graceful_shutdown=1`. Using a FastAPI lifespan context manager, I do not get to the shutdown event. This happens on my laptop running Windows 11 even with something as small as:
```
from contextlib import asynccontextmanager

import uvicorn
from fastapi import FastAPI


@asynccontextmanager
async def lifespan(app: FastAPI):
    yield
    print("test123") # <-- We only get here after minutes of waiting


app = FastAPI(lifespan=lifespan)
if __name__ == "__main__":
    uvicorn.run("module:app", reload=True, timeout_graceful_shutdown=1)
```

**Comment by Kludex** [comment, marked as answer]
- I've created https://github.com/encode/uvicorn/issues/2000 from this discussion.

**Reply by 3to1null** [reply]
Wow! Quick response! Is there any information that I could provide that might help narrowing this issue down?

**Reply by Kludex** [reply]
Yes. You can check the 2 PRs related to Windows that were merged for 0.22.0, try to revert in a branch, and use that branch locally and see if it solves the problem.

**Comment by trim21** [comment]
Find this also happens in powershell in windows terminal, too
